electron_app/src/spell-check.js

```diff
@@ -0,0 +1,309 @@
+	let directory = null;
+    if (app.getAppPath().indexOf('app.asar') > -1) {
+		directory = jetpack.cwd(
+			app.getAppPath().replace('/app.asar', ''),
+			'dictionaries')
+	} else {
+		directory = jetpack.cwd(
+			app.getAppPath(),
+			'electron_app',
+			'dictionaries')
+	}
+	console.log('dir', directory.toString());
+	const found = await directory.findAsync({ matching: '*.{aff,dic}' });
+	const installedDictionaries = found.map((fileName) => path.basename(fileName, path.extname(fileName)));
+	this.dictionariesPath = directory.path();
+	this.dictionaries = Array.from(new Set([ ...embeddedDictionaries, ...installedDictionaries ])).sort();
+	this.isMultiLanguage = embeddedDictionaries.length > 0 && process.platform !== 'win32';
+};
+
+const updateChecker = () => {
+	try {
+		if (this.enabledDictionaries.length === 0) {
+			this.checker = () => true;
+			return;
+		}
+
+		if (this.enabledDictionaries.length === 1) {
+			let enabled = false;
+			this.checker = mem((text) => {
+				if (!enabled) {
+					spellchecker.setDictionary(this.enabledDictionaries[0], this.dictionariesPath);
+					enabled = true;
+				}
+				return !spellchecker.isMisspelled(text);
+			});
+			return;
+		}
+
+		const singleDictionaryChecker = mem(
+			((dictionariesPath, dictionary, text) => {
+				spellchecker.setDictionary(dictionary, dictionariesPath);
+				return !spellchecker.isMisspelled(text);
+			}).bind(null, this.dictionariesPath)
+		);
+
+		this.checker = mem(
+			((dictionaries, text) => dictionaries.some((dictionary) => singleDictionaryChecker(dictionary, text))).bind(
+				null,
+				this.enabledDictionaries
+			)
+		);
+	} finally {
+		webFrame.setSpellCheckProvider('', false, { spellCheck: this.checker });
+	}
+};
+
+const enable = (...dictionaries) => {
+	dictionaries = filterDictionaries(dictionaries);
+	if (this.isMultiLanguage) {
+		this.enabledDictionaries = [ ...this.enabledDictionaries, ...dictionaries ];
+	} else {
+		this.enabledDictionaries = [ dictionaries[0] ];
+	}
+
+	localStorage.setItem('spellcheckerDictionaries', JSON.stringify(this.enabledDictionaries));
+
+	updateChecker();
+	return this.enabledDictionaries.length > 0;
+};
+
+const disable = (...dictionaries) => {
+	dictionaries = filterDictionaries(dictionaries);
+
+	this.enabledDictionaries = this.enabledDictionaries.filter((dictionary) => !dictionaries.includes(dictionary));
+	localStorage.setItem('spellcheckerDictionaries', JSON.stringify(this.enabledDictionaries));
+
+	updateChecker();
+};
+
+const setDefaultEnabledDictionaries = () => {
+	const selectedDictionaries = (() => {
+		try {
+			const enabledDictionaries = JSON.parse(localStorage.getItem('spellcheckerDictionaries'));
+			return Array.isArray(enabledDictionaries) ? enabledDictionaries.map(String) : null;
+		} catch (error) {
+			console.error(error);
+			return null;
+		}
+	})();
+
+	if (selectedDictionaries) {
+		enable(...selectedDictionaries);
+		return;
+	}
+
+	const userLanguage = localStorage.getItem('userLanguage');
```

Comment:
About this whole block: Can't you use the language the User selected inside the app using the already existing code?

Reply:
I'll try ;)